src/lib/optimizer/strategy/like_replacement_rule.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "abstract_rule.hpp"
+#include "expression/expression_functional.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * This optimizer rule replaces PredicateNodes with a like condition by BinaryPredicateExpressions for some special cases.
```

Comment:
"replaces ... with", not "replaces ... by", right?

Reply:
I thought `replace` could be used with both prepositions, `with` and `by`.

Reply:
just sounded weird to me, not about to look up english grammar rules.

Reply:
There used to be a page where you could see the most frequently used prepositions for a verb, but I can't find it anymore. This suggests that 'with' is more common: 
https://english.stackexchange.com/a/151352

Reply:
In this case it could even be understood that `PredicateNodes` are replaced with `a like condition`. So maybe `PredicateNodes that contain a like condition`